pandas/core/series.py

```diff
@@ -2077,8 +2077,11 @@ def __rmatmul__(self, other):
     def searchsorted(self, value, side='left', sorter=None):
         if sorter is not None:
             sorter = ensure_platform_int(sorter)
-        return self._values.searchsorted(Series(value)._values,
-                                         side=side, sorter=sorter)
+        if not is_extension_type(self._values):
+            value = np.asarray(value, dtype=self._values.dtype)
+            value = value[..., np.newaxis] if value.ndim == 0 else value
```

Comment:
Without this shim ``searchsorted`` returns a scalar, so this line is only to ensure that - like in master - an 1-dim array is returned.

I like better the numpy convention of returning a scalar from searchsorted when possible, but that can be for another PR.

Reply:
can you add a comment here on why you are doing this (and the expl above as well)